src/main/java/dev/openfeature/sdk/ProviderRepository.java

```diff
@@ -55,6 +55,18 @@ public boolean isDefaultProvider(FeatureProvider provider) {
         return this.getProvider().equals(provider);
     }
 
+    private void setProvider(FeatureProvider provider,
+            Consumer<FeatureProvider> afterSet,
+            Consumer<FeatureProvider> afterInit,
+            Consumer<FeatureProvider> afterShutdown,
+            BiConsumer<FeatureProvider, String> afterError,
+            boolean waitForInit) {
+        if (provider == null) {
+            throw new IllegalArgumentException("Provider cannot be null");
+        }
+        initializeProvider(null, provider, afterSet, afterInit, afterShutdown, afterError, waitForInit);
+    }
```

Comment:
This is a package private class, so we _could_ also just have one constructor, and update calls to use that, passing nulls/defaults where needed. I don't mind either way though.

Reply:
cool, then no need to worry about breaking changes on public methods, I'll update it.

So in this case of package private class, the methods can be be changed to non-public as well, right ?

Reply:
> So in this case of package private class, the methods can be be changed to non-public as well, right ?

In Java, I'm of the opinion that with package private classes, certain methods should still be public. This makes it clear which ones are intended to be used from the outside. This is a style choice though.